### ISSUE TITLE
Need Help, Have Problems on Training Model C 

### ISSUE THREAD

**Issue by qiexing** [AlfredXiangWu/face_verification_experiment#77]
Hi, it's amazing to see the great performance your network achieves. 
So i want to train the model.
I follow the procedure you mention. 1. detect face. 2. crop and transform to gray image.
but with out face alignment.
But, the training error seems not decrease. 
I1203 12:57:21.692827 16815 solver.cpp:404]     Test net output #0: accuracy = 0
I1203 12:57:21.692917 16815 solver.cpp:404]     Test net output #1: loss = 9.80349 (* 1 = 9.80349 loss)
I1203 12:57:21.817539 16815 solver.cpp:228] Iteration 0, loss = 11.026
I1203 12:57:21.817608 16815 solver.cpp:244]     Train net output #0: loss = 11.026 (* 1 = 11.026 loss)
I1203 12:57:21.817652 16815 sgd_solver.cpp:106] Iteration 0, lr = 0.003
I1203 12:57:43.937072 16815 solver.cpp:228] Iteration 100, loss = 9.31884
I1203 12:57:43.937497 16815 solver.cpp:244]     Train net output #0: loss = 9.31884 (* 1 = 9.31884 loss)
I1203 12:57:43.937536 16815 sgd_solver.cpp:106] Iteration 100, lr = 0.003
I1203 12:58:06.112326 16815 solver.cpp:228] Iteration 200, loss = 9.31977
I1203 12:58:06.112411 16815 solver.cpp:244]     Train net output #0: loss = 9.31977 (* 1 = 9.31977 loss)
I1203 12:58:06.112428 16815 sgd_solver.cpp:106] Iteration 200, lr = 0.003
I1203 12:58:28.295809 16815 solver.cpp:228] Iteration 300, loss = 9.27393
I1203 12:58:28.295948 16815 solver.cpp:244]     Train net output #0: loss = 9.27393 (* 1 = 9.27393 loss)
I1203 12:58:28.295964 16815 sgd_solver.cpp:106] Iteration 300, lr = 0.003
I1203 12:58:50.493698 16815 solver.cpp:228] Iteration 400, loss = 9.28893
I1203 12:58:50.493779 16815 solver.cpp:244]     Train net output #0: loss = 9.28893 (* 1 = 9.28893 loss)
I1203 12:58:50.493794 16815 sgd_solver.cpp:106] Iteration 400, lr = 0.003
I1203 12:59:12.356719 16815 solver.cpp:228] Iteration 500, loss = 9.29393
I1203 12:59:12.356961 16815 solver.cpp:244]     Train net output #0: loss = 9.29393 (* 1 = 9.29393 loss)
I1203 12:59:12.356982 16815 sgd_solver.cpp:106] Iteration 500, lr = 0.003
I1203 12:59:33.527235 16815 solver.cpp:228] Iteration 600, loss = 9.26926
I1203 12:59:33.527314 16815 solver.cpp:244]     Train net output #0: loss = 9.26926 (* 1 = 9.26926 loss)
I1203 12:59:33.527333 16815 sgd_solver.cpp:106] Iteration 600, lr = 0.003

The training error is still  9.16429 after 3500 iterations.
I put the prototxts and sample images on :
https://www.dropbox.com/sh/jag6r3v3y5y64uj/AABmZ02e-xm5DKyFrUplUtyxa?dl=0

If there is something wrong in my setting? .

**Comment by Martyn10** [comment]
Edit: Sorry just re-read you're post where you mention you do it without face alignment. Have you tried this with other models before and was it successful?

I've only looked at your sample images so not sure if there is something wrong elsewhere.
If you look at section 4 of Wu's paper, the training images should be rotated so the eyes are horizontally level. The center of eyes should be horizontally centered and be 48px from the top of the image.
The distance between the mid point of eyes and the mid point of mouth should also be 48px.
See Figure 2 in paper for an example.

**Comment by qiexing** [comment]
@Martyn10  Hi, Martyn, Yes, i use this data to train the seata face: https://github.com/seetaface/SeetaFaceEngine。
and the training accuracy can decrease to 0.2 and get around 84% test accuracy.
I think the no alignment data maybe can work.
So i wonder is there if something wrong in my model config.

**Comment by Martyn10** [comment]
In section 4.5 Wu mentions training network using the CASIA database. Then the learning rate for convolution layers is set to 0 and the FC layers are fine-tuned using the MSCELEB database. Then the learning rate for convolutional layers is adjusted again. I'm not sure if pre-training with CASIA is necessary or not though.
The learning rate for fc2 layer is also different.

**Comment by qiexing** [comment]
@Martyn10  Thank you, maybe

**Comment by qiexing** [comment]
@Martyn10 Thank you,  I try to train the Model B with CASIA data instead of C.
